src/test/java/com/cloudant/tests/ViewsTest.java

```diff
@@ -107,13 +107,227 @@ public void byStartAndEndKey() {
+     */
+    @Test
+    public void byStartAndEndIntKeyArray() {
+        init();
+
+        View query = db.view("example/created")
+                .startKey(1)
+                .endKey(2000);
+
+        List<Object> result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+
+        Page page = null;
+        try {
+            page = query.queryPage(30, null, Object.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        assertThat(page.getResultList().size(), is(3));
+
+        query = db.view("example/created")
+                .startKey(new Object[]{1})
+                .endKey(new Object[]{2000});
+
+
+        result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+    }
+
+    /**
+     * Assert that passing a key with two values in query
+     * and queryView will produce a result list of all docs.
+     *
+     * Run the same assert above but with a key that is an
+     * object array containing only numbers.
+     */
+    @Test
+    public void byStartAndEndTwoKeyArray() {
+        init();
+
+        View query = db.view("example/creator_created")
+                .startKey("uuid", 1)
+                .endKey("uuid", 2000);
+
+        List<Object> result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+
+        Page page = null;
+        try {
+            page = query.queryPage(30, null, Object.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        assertThat(page.getResultList().size(), is(3));
+
+
+        query = db.view("example/creator_created")
+                .startKey(new Object[]{"uuid", 1})
+                .endKey(new Object[]{"uuid", 2000});
+
+
+        result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+
+        //Key array with only numbers
+        query = db.view("example/created_total")
+                .startKey(1, 12.00)
+                .endKey(2000, 15.00);
+
+
+        result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+
+        page = null;
+        try {
+            page = query.queryPage(30, null, Object.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        assertThat(page.getResultList().size(), is(3));
+    }
+
+    /**
+     * Assert that passing a key with three different values in
+     * query and queryView will produce a result list of all docs.
+     *
+     * Run the same assert above but with a key that is an object array
+     * and should only return a result with 'true' as the first value.
+     */
+    @Test
+    public void byStartAndEndThreeKeyArray() {
+        init();
+
+        View query = db.view("example/boolean_creator_created")
+                .startKey(false, "uuid", 1)
+                .endKey(false, "uuid", 2000);
+
+        List<Object> result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
```

Comment:
I've changed the start/end key to produce different results in all the multi-value key tests in e75a2f1.  What other test results should we check for?